pkg/operator/controllers/machineset/machineset_controller.go

```diff
@@ -54,9 +56,20 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	}
 
 	r.log.Debug("running")
+
+	cnds, err := conditions.GetControllerConditions(ctx, r.client, ControllerName)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
 	modifiedMachineset := &machinev1beta1.MachineSet{}
 	err = r.client.Get(ctx, types.NamespacedName{Name: request.Name, Namespace: machineSetsNamespace}, modifiedMachineset)
```

Comment:
There are a few additional CRUD operations in this controller as well as this one (list the machinesets, update the machinesets' replicas). Should we also be emitting the same statuses / messages if we get errors there?

Reply:
https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.6/pkg/client/fake There does not currently appear to be a way to simulate any errors in the fake client, so adding test coverage for operations like List or Update that would not ever fail due to the presence/absence of resource is not possible.